VisualBasic.NET Build and Debug in VS2015

smaillet-ms opened the pull request:
Fixes #387 
- updated VisualBasic Targets file to set required properties
- Updated VB project templates to disable optimization for debug builds so binaries are debuggable
- Updated the VSProject system tp check for explicit mscorlib assembly reference and if not present to add it to the internal list of assemblies for deployment.
- Added use of FLAVOR_SDK environment variable to build_sdk.cmd so that the SDK components can be built for debug configurations. (This helps when debugging the VS integration and MFDeploy)

smaillet-ms committed:
- Added SetupVBRuntime Target to force the VB runtime and set the "MyType" property so that netmfvb projects can build and deploy properly

smaillet-ms committed:
- Added Debug configuration suport to WinUSB to allow building Debug
- Fixed VB project templates to set Optimized property to false in debug configurations to enable debugging
- Modified VSProject.cs to test for mscorlib as an explicit dependency and if not present, add it so that various versions of VS and MSBUild common project systems work correctly.
- Added optional flavor flag in build_sdk to allow building the SDK as Debug. This helps in debugging working with the VS integration in debug mode as it builds all the dependencies in one pass.

miloush commented on Framework/CorDebug/VsProject.cs:
```diff
@@ -142,11 +143,24 @@ public string[] GetDependencies( bool fIncludeStartProgram, bool fPE, bool fTarg
+                // false so it won't show in the IDE. By testing for it and adding it if not found
+                // both cases are covered. 
+                var q = from path in table.OfType<string>( )
+                        where 0 == string.Compare( Path.GetFileName( path ), "mscorlib.dll", StringComparison.InvariantCultureIgnoreCase )
```
I believe this should be [OrdinalIgnoreCase](https://msdn.microsoft.com/en-us/library/ms973919.aspx) comparison, invariant culture considers things equal that file system does not. And you can use `string.Equals(string, string, StringComparison)` for better readability.

miloush commented on build_sdk.cmd:
```diff
@@ -9,6 +9,8 @@ if "%1" == "-?" goto :ShowUsage
 if /i "%1" == "/h" goto :ShowUsage
 if /i "%1" == "-h" goto :ShowUsage
 
+if "%FLAVOR_SDK%" == "" FLAVOR_SDK=RTM
```
This probably was meant to be **set** FLAVOR_SDK=RTM

miloush commented:
:+1: works for me

smaillet-ms committed:
- FIxed issues from Code review
- added Readme.md for the crypto libraries as the readme was left out from the original commit for the crypto libs
